expression/builtin_string.go

```diff
@@ -904,83 +905,207 @@ type locateFunctionClass struct {
 }
 
 func (c *locateFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinLocateSig{newBaseBuiltinFunc(args, ctx)}
-	return sig.setSelf(sig), errors.Trace(c.verifyArgs(args))
+	argsErr := c.verifyArgs(args)
+	var (
+		bf  baseBuiltinFunc
+		err error
+	)
+
+	hasStartPos := len(args) == 3
+	caseSensitive := types.IsBinaryStr(args[0].GetType()) || types.IsBinaryStr(args[1].GetType())
```

Comment:
I prefer 'caseSensitive' because it demonstrates the meaning of this variable directly. Maybe I should add a comment here to clarify the relationship between 'case-sensitive' and ' has binary input'